Submit

tsohleDev opened the pull request:
Create the App with Pokemon API and the Involvement API
- Create Home Page
- Comment Pop Up
- Reservations Pop Up

tsohleDev committed:
Configure webpack and jest

>webpack to accept scss
>jest for testing and mocking css

ReemMohamedAbdelfatah committed:
Try Consume PokeApi

ReemMohamedAbdelfatah committed:
jest set up

tsohleDev committed:
Add pop up window implementation

tsohleDev committed:
Unit test the create elements function

tsohleDev committed:
Merge branch 'development' into 14-4pt-display-comments-pop-up-with-selected-items-details-student-b

tsohleDev committed:
Change html Title

tsohleDev committed:
Merge branch '14-4pt-display-comments-pop-up-with-selected-items-details-student-b' of https://github.com/ReemMohamedAbdelfatah/Capstoneproject-moduleJS into 14-4pt-display-comments-pop-up-with-selected-items-details-student-b

tsohleDev committed:
Change html Title

tsohleDev committed:
Ahh heading in page

ReemMohamedAbdelfatah committed:
Add Header,Footer and complete list style

ReemMohamedAbdelfatah committed:
Finish styling

ReemMohamedAbdelfatah committed:
fix linter errors

tsohleDev committed:
Call Involvement to get comments

tsohleDev committed:
Involvement POST form

tsohleDev committed:
fix lints

tsohleDev committed:
Impliment counter

tsohleDev committed:
FIx lints

tsohleDev committed:
mock the API

tsohleDev committed:
Test the comment counter module

tsohleDev committed:
Remove lints

tsohleDev committed:
Add content for the pokemon

ReemMohamedAbdelfatah committed:
add fetch likes and post like from api

tsohleDev committed:
include Pokemon Api

tsohleDev committed:
reserve module

tsohleDev committed:
Reservations

ReemMohamedAbdelfatah committed:
Add test and counter

ReemMohamedAbdelfatah committed:
Merge branch 'Reem-branch' of into Reem-branch

tsohleDev committed:
Merge branch 'development' of https://github.com/ReemMohamedAbdelfatah/Capstoneproject-moduleJS into development

tsohleDev committed:
Attach to home page

tsohleDev committed:
Merge branch '2-3pt-add-tests-for-comments-counter-student-b' into reservations

tsohleDev committed:
Attach to home page

tsohleDev committed:
Remove linters

tsohleDev committed:
Add checks

tsohleDev committed:
Fix bug

tsohleDev committed:
Fix bug II

tsohleDev committed:
Add README

tsohleDev committed:
Update README.md

tsohleDev committed:
Include Tests

tsohleDev committed:
Merge branch 'Submit' of https://github.com/ReemMohamedAbdelfatah/Capstoneproject-moduleJS into Submit

Ghiftee commented on src/index.js:
```diff
@@ -0,0 +1,87 @@
+/* eslint-disable  no-unused-vars  , no-use-before-define , no-plusplus */
```
- [x] I suggest that you do not disable linters as they help us write cleaner code.

Ghiftee commented on src/temp.html:
```diff
@@ -1,12 +1,34 @@
+    />
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+  </head>
+  <body>
```
- [x] You currently have two HTML files, kindly remove one

tsohleDev replied:
I think this is the same file, just that the red part was removed

Ghiftee commented on README.md:
```diff
@@ -70,132 +68,167 @@
-👤 **Reem Muhammed**
+<img src="/images/github.svg" alt="logo" width="18"  height="18" />[githubhandle](https://github.com/ReemMohamedAbdelfatah)
+<br>
+<img src="/images/linkedin.svg" alt="logo" width="18"  height="18" />[linkedIn](https://www.linkedin.com/in/reem-abd-el-fatah-a07543116)
 
-- GitHub: [@github](https://github.com/ReemMohamedAbdelfatah)
-- LinkedIn: [LinkedIn](https://www.linkedin.com/in/reem-abd-el-fatah-a07543116)
+<h4>Tsohle Mokhemisi </h4>
+
+<img src="/images/github.svg" alt="logo" width="18"  height="18" />[githubhandle](https://github.com/tsohleDev2)
+<br>
+<img src="/images/twitter.svg" alt="logo" width="18"  height="18" /> [twitterhandle](https://twitter.com/RealTsohle)
+<br>
+<img src="/images/linkedin.svg" alt="logo" width="18"  height="18" />[linkedIn](https://www.linkedin.com/in/tsohle-mokhemisi-3687401b2/)
```
- [x] These images are all broken and this gives your README an unprofessional look. Kindly fix it.

Ghiftee commented on tests/homeCounter.test.js:
```diff
@@ -0,0 +1,63 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import counter from '../src/modules/counter.js';
+import displayPokemon from '../__mocks__/likeAPiMock.js';
+
```
- [x] For better readability of your code, I suggest that you put all the tests in one folder instead of creating two folders for tests.

Ghiftee commented on dist/main.4cd43ce0f8cc44fc4588.js:
```diff
@@ -0,0 +1,767 @@
+/******/ (() => { // webpackBootstrap
```
- [x] When naming a file, adding numbers is not best practice so I suggest you remove the numbers from the name of this file

Ghiftee commented on src/tests/commentCounter.test.js:
```diff
@@ -0,0 +1,41 @@
+  });
+});
+
+describe('conting elements', () => {
+  test('nothing is rendered but API returned data', () => {
+    const div = document.createElement('div');
+
+    appendCommentsMock(div, null);
+    const count = PopUp.countComments(div);
+    expect(count).toBe(0);
+  });
+});
```
- [x] Good job creating tests for the comment counter, kindly include tests for the like counter that covers all scenarios e.g when there are 0 likes

Ghiftee reviewed (changes requested):
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @ReemMohamedAbdelfatah and @tsohleDev  👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ You documented your work professionally
✔️ Your tests pass
✔️ There are no linter errors

However, there are some issues that you need to work on to go to the next project but you are almost there!

![](https://media.giphy.com/media/3og0IPMeREHpEV0f60/giphy.gif)

## Required Changes ♻️
_Check the comments under the review._

- [x] Kindly ensure that you follow the layout of the given design as this is a requirement.

### Given layout
<img width="489" alt="Screenshot 2023-01-07 at 13 11 50" src="https://user-images.githubusercontent.com/64774936/211150085-7b48e84c-9a39-4f79-a252-ecb20e586a7c.png">

### Your layout
<img width="1423" alt="Screenshot 2023-01-07 at 13 12 00" src="https://user-images.githubusercontent.com/64774936/211150096-0cab6f0b-dfa7-4d36-8e28-6b7a61bc892b.png">

- [x] When I like an item, the likes get updated immediately, good job! However, if I reload the page, the likes get decreased by 1. Kindly fix this

### Before reload

<img width="273" alt="Screenshot 2023-01-07 at 13 01 23" src="https://user-images.githubusercontent.com/64774936/211149918-9739fa38-7994-4a87-bc1c-78dcf40b6f14.png">

### After reload
<img width="278" alt="Screenshot 2023-01-07 at 13 01 29" src="https://user-images.githubusercontent.com/64774936/211149939-3047eddc-a094-4c28-9eb6-ecdc7397d3bb.png">

- [x] I cannot create a comment. When I click on the comment button, the pop-up for creating a comment does not appear. Kindly fix this as it is a part of the requirement.
- [x] In order to make your README more personalized, I suggest you remove the Microverse logo from your README and replace it with yours if you have one. If you do not have one, you can leave the README without a logo
- [x] Your ```src/modules/HomePage.js``` file is empty, please remove it from your codebase.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

tsohleDev committed:
Update README.md

tsohleDev committed:
Fix bug
Like button to load
Remove logo from README and Broken pictures
Fix linters
Fix Layout

tsohleDev committed:
Fix bug
Like button to load
Remove logo from README and Broken pictures
Fix linters
Fix Layout

1GORDON commented on README.md:
```diff
@@ -70,132 +63,167 @@
+
+<h4>Tsohle Mokhemisi </h4>
+
+[github](https://github.com/tsohleDev2)
```
Great job team 🤝

- [ ] [Optional] Kindly fix the broken link for the second author for the GitHub link. This will make your readme file more professional.

tsohleDev replied:
Thnak you sir

1GORDON reviewed (approved):
# Status: APPROVED 🟢 

Hi @tsohleDev & @ReemMohamedAbdelfatah ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ReemMohamedAbdelfatah commented:
@1GORDON Thank you!
